src/main/java/com/cinema/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.cinema.shop.service.impl;
+
+import com.cinema.shop.dao.ShoppingCartDao;
+import com.cinema.shop.dao.TicketDao;
+import com.cinema.shop.lib.Inject;
+import com.cinema.shop.lib.Service;
+import com.cinema.shop.model.MovieSession;
+import com.cinema.shop.model.ShoppingCart;
+import com.cinema.shop.model.Ticket;
+import com.cinema.shop.model.User;
+import com.cinema.shop.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.getTickets().add(ticket);
```

Comment:
I'm not sure, but you add ticket, which has not been assosiated with DB yet. Probably, at first you should add it with ticketDao and then add to shopping cart. But I may be wrong

Reply:
> I'm not sure, but you add ticket, which has not been assosiated with DB yet. Probably, at first you should add it with ticketDao and then add to shopping cart. But I may be wrong

\+